apps/rush-lib/src/logic/policy/ShrinkwrapFilePolicy.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as colors from 'colors';
+import * as os from 'os';
+import { RushConfiguration } from '../../api/RushConfiguration';
+import { AlreadyReportedError } from '../../utilities/AlreadyReportedError';
+import { IPolicyValidatorOptions } from './PolicyValidator';
+import { BaseShrinkwrapFile } from '../base/BaseShrinkwrapFile';
+import { ShrinkwrapFileFactory } from '../ShrinkwrapFileFactory';
+
+/**
+ *  A policy that prevents manual changes to shrinkwrap files used by package managers. This policy can
+ *  be enabled by setting PnpmOptionsConfiguration.preventManualShrinkwrapChanges to true.gul
+ */
+export class ShrinkwrapFilePolicy {
+  public static validate(rushConfiguration: RushConfiguration, options: IPolicyValidatorOptions): void {
+    // If allowShrinkwrapUpdates is true, then we're updating the shrinkwrap and cannot validate
+    // the hash. Additionally, we need to check that the feature is enabled at all.
+    if (options.allowShrinkwrapUpdates || !ShrinkwrapFilePolicy.isEnabled(rushConfiguration)) {
+      return;
+    }
+
+    console.log('Checking shrinkwrap file hash to ensure no invalid changes were made.' + os.EOL);
+
+    // Since we know it is PNPM specific, we can assume the shrinkwrap type.
+    const shrinkwrapFile: BaseShrinkwrapFile | undefined = ShrinkwrapFileFactory.getShrinkwrapFile(
+      rushConfiguration.packageManager,
+      rushConfiguration.getCommittedShrinkwrapFilename(options.shrinkwrapVariant)
+    );
+
+    if (!shrinkwrapFile) {
+      throw new Error('Could not load the shrinkwrap file!');
```

Comment:
Test the case where `rush install` or `rush update` is invoked when the shrinkwrap file does not exist.  Is there an error?  (Should there be?)  If there is an error, is the message easy to understand?

Reply:
Probably shouldn't fail in this scenario. Modified it to skip validation if the shrinkwrap can't be found.